src/gsi/setuppm2_5.f90

```diff
@@ -669,17 +736,43 @@ subroutine setuppm2_5(obsLL,odiagLL,lunin,mype,nreal,nobs,isis,is,conv_diagsave)
            call tintrp2a11(ges_pm2_5,pm2_5ges,dlat,dlon,dtime,hrdifsig,&
                 mype,nfldsig)
            innov = conc - pm2_5ges
+           if (laeroana_fv3smoke) then
+              if ( -1.0*innov >= pm2_5_innov_threshold .or. & 
+                  (innov > pm2_5_innov_threshold .and. pm2_5ges >=1.0_r_kind).or. &
+                  (conc >= 40.0_r_kind .and. pm2_5ges >=1.0_r_kind).or. &
+                   conc >= 100.0_r_kind ) then
+                 innov = innov
+              else 
+                 innov = 0.0_r_kind 
+              end if
+              if (tv_ges-273.15_r_kind < 0.0_r_kind) then
+                 innov = 0.0_r_kind
```

Comment:
no innovations below freezing?

Reply:
RRFS-SD only has one smoke and one dust tracer. So that we only can assimilate the smoke/dust dominated PM2.5. However, this is a challenge to identify which station is smoke/dust affected from the value itself. In addition to ue of smoke/dust mask from satellite in obs preprocess, here a physical constraint of temperature is used to exclude non-smoke affected  PM2.5 obs since the tempature should be greater than 0.0 where fire takes place. We may adjust this in future after more tests.